tests/devices/i04/test_transfocator.py

```diff
@@ -1,66 +1,75 @@
+import asyncio
+from unittest.mock import AsyncMock, patch
 
 import pytest
-from ophyd.sim import make_fake_device
-from ophyd.status import SubscriptionStatus
+from ophyd_async.core import (
+    DeviceCollector,
+    set_mock_value,
+    wait_for_value,
+)
 
 from dodal.devices.i04.transfocator import Transfocator
 
 
 @pytest.fixture
-def fake_transfocator():
-    FakeTransfocator = make_fake_device(Transfocator)
-    transfocator: Transfocator = FakeTransfocator(name="test_transfocator")
-    yield transfocator
+async def fake_transfocator() -> Transfocator:
+    async with DeviceCollector(mock=True):
+        transfocator = Transfocator(prefix="", name="transfocator")
+    return transfocator
 
 
 def given_predicted_lenses_is_half_of_beamsize(transfocator: Transfocator):
     def lens_number_is_half_beamsize(value, *args, **kwargs):
-        transfocator.predicted_vertical_num_lenses.sim_put(int(value / 2))  # type: ignore
+        set_mock_value(transfocator.predicted_vertical_num_lenses, int(value / 2))
 
-    transfocator.beamsize_set_microns.subscribe(lens_number_is_half_beamsize)
+    transfocator.beamsize_set_microns.subscribe_value(lens_number_is_half_beamsize)
 
 
-def test_given_beamsize_already_set_then_when_transfocator_set_then_returns_immediately(
+async def set_beamsize_to_same_value_as_mock_signal(
+    transfocator: Transfocator, value: float
+):
+    transfocator.set(value)
+    set_mock_value(transfocator.beamsize_set_microns, value)
+
+
+async def test_given_beamsize_already_set_then_when_transfocator_set_then_returns_immediately(
     fake_transfocator: Transfocator,
 ):
-    fake_transfocator.beamsize_set_microns.sim_put(100)  # type: ignore
-    set_status = fake_transfocator.set(100)
-    set_status.wait(0.01)
-    assert set_status.done and set_status.success
+    await asyncio.wait_for(
+        set_beamsize_to_same_value_as_mock_signal(fake_transfocator, 100.0),
+        timeout=0.01,
+    )
 
 
-@patch("dodal.devices.i04.transfocator.sleep")
-def test_when_beamsize_set_then_set_correctly_on_device_and_waited_on(
-    mock_sleep,
+async def test_when_beamsize_set_then_set_correctly_on_device_and_waited_on(
     fake_transfocator: Transfocator,
 ):
     given_predicted_lenses_is_half_of_beamsize(fake_transfocator)
 
-    transfocator_statuses = SubscriptionStatus(
-        fake_transfocator.predicted_vertical_num_lenses,
-        lambda *, old_value, value, **kwargs: value == 157,
-    )
-    transfocator_statuses &= SubscriptionStatus(
-        fake_transfocator.number_filters_sp,
-        lambda *, old_value, value, **kwargs: value == 157,
-    )
-    transfocator_statuses &= SubscriptionStatus(
-        fake_transfocator.start,
-        lambda *, old_value, value, **kwargs: value == 1,
-    )
-
+    fake_transfocator.start_rbv.get_value = AsyncMock(side_effect=[1, 0])
     set_status = fake_transfocator.set(315)
 
-    transfocator_statuses.wait(0.1)
     assert not set_status.done
```

Comment:
This assertion isn't useful anymore, since we haven't awaited on `set_status`, so there's no chance of it being done at this stage. 

In one of my earlier comments I suggested using `asyncio.create_task` in the `set` method. If we do this, then we should be able to do the same kind of assertion here

Reply:
As discussed, this assert statement would be useful if there were any awaits inbetween this and the `set` being called. Eg an `asyncio.sleep`